kong/api/routes/upstreams.lua

```diff
@@ -40,7 +42,48 @@ return {
     end,
 
     GET = function(self, dao_factory)
-      crud.paginated_set(self, dao_factory.targets)
+      -- by default, just get a paginated list
+      if not self.params.active then
+        return crud.paginated_set(self, dao_factory.targets)
+      end
+
+      self.params.active = nil
+
+      -- user wanted a list of all active targets for this upstream
+      local target_history, err = dao_factory.targets:find_all({
+        upstream_id = self.params.upstream_id,
+      })
+
+      if not target_history then
+        return app_helpers.yield_error(err)
+      end
+
+      -- sort and walk based on target and creation time
+      for _, target in ipairs(target_history) do
+        target.order = target.target..":"..target.created_at
+      end
+      table.sort(target_history, function(a, b) return a.order > b.order end)
```

Comment:
this sorting is not ok, see https://github.com/Mashape/kong/blob/6ca88e4d5732320d03cf7b4b40e7f79c903392a7/kong/api/routes/upstreams.lua#L99-L101

It has to be deterministic, so the same order must be used everywhere. As the timestamp might have duplicates (edge case, but could happen) and then the sort order is relevant.

`target.order = target.target..":"..target.created_at..":"..target.id` ought to be fine